examples/dynamic-bridge-app/linux/main.cpp

```diff
@@ -176,6 +189,141 @@ std::unique_ptr<GeneratedCluster> CreateCluster(chip::ClusterId id)
+}
+void ReadValueFromBuffer(chip::TLV::TLVWriter & wr, const char * buffer, const float *)
+{
+    wr.Put(chip::TLV::Tag(), (float) atof(buffer));
+}
+void ReadValueFromBuffer(chip::TLV::TLVWriter & wr, const char * buffer, const double *)
+{
+    wr.Put(chip::TLV::Tag(), atof(buffer));
+}
+void ReadValueFromBuffer(chip::TLV::TLVWriter & wr, const char * buffer, const int64_t *)
+{
+    wr.Put(chip::TLV::Tag(), (int64_t) strtoll(buffer, nullptr, 10));
+}
+void ReadValueFromBuffer(chip::TLV::TLVWriter & wr, const char * buffer, const uint64_t *)
+{
+    wr.Put(chip::TLV::Tag(), (uint64_t) strtoll(buffer, nullptr, 10));
+}
+void ReadValueFromBuffer(chip::TLV::TLVWriter & wr, const char * buffer, const bool *)
+{
+    wr.PutBoolean(chip::TLV::Tag(), (*buffer) ? true : false);
+}
+
+template <typename T>
+void ReadValueFromBuffer(chip::TLV::TLVWriter & wr, const char * buffer, const T *)
+{
+    wr.Put(chip::TLV::Tag(), (int64_t) strtoll(buffer, nullptr, 10));
+}
+
+void WriteValueToBuffer(const bool & value, uint8_t * buffer, const uint16_t & maxReadLength)
+{
+    if (maxReadLength == 1)
+        *buffer = value ? 1 : 0;
+}
+
+template <typename T>
+void WriteValueToBuffer(const T & value, uint8_t * buffer, const uint16_t & maxReadLength)
+{
+    size_t value_size = sizeof(value);
+    if (maxReadLength >= value_size)
+    {
+        memcpy(buffer, &value, value_size);
+    }
+}
+
+template <typename T>
+void WriteValueToBuffer(chip::TLV::TLVReader & reader, uint8_t * buffer, const uint16_t & maxReadLength)
+{
+    T v;
+    chip::app::DataModel::Decode(reader, v);
+    WriteValueToBuffer(v, buffer, maxReadLength);
+}
+
+// if `read` is true, read from `buffer`, and write to `data`
+// otherwise, read from `data`, and write to `buffer`
+template <typename T>
+void ReadOrWriteBuffer(std::vector<uint8_t> * data, uint8_t * buffer, uint16_t maxReadLength, bool read)
+{
+    if (read)
+    {
+        chip::TLV::TLVWriter wr;
+        wr.Init(data->data(), data->size());
+        ReadValueFromBuffer(wr, (char *) buffer, (const T *) nullptr);
+        wr.Finalize();
+        data->resize(wr.GetLengthWritten());
+    }
+    else
+    {
+        chip::TLV::TLVReader rd;
+        rd.Init(data->data(), data->size());
+        rd.Next();
+        WriteValueToBuffer<T>(rd, buffer, maxReadLength);
+    }
+}
+
+void ReadOrWriteBufferForType(std::vector<uint8_t> * data, uint8_t * buffer, EmberAfAttributeType type, uint16_t maxReadLength,
+                              bool read)
+{
+    switch (type)
+    {
+    case ZCL_OCTET_STRING_ATTRIBUTE_TYPE:
+    case ZCL_CHAR_STRING_ATTRIBUTE_TYPE:
+        ReadOrWriteBuffer<Span<const char>>(data, buffer, maxReadLength, read);
+        break;
+    case ZCL_LONG_OCTET_STRING_ATTRIBUTE_TYPE:
+    case ZCL_LONG_CHAR_STRING_ATTRIBUTE_TYPE:
+        ReadOrWriteBuffer<ByteSpan>(data, buffer, maxReadLength, read);
+        break;
+    case ZCL_STRUCT_ATTRIBUTE_TYPE:
+        // structs not supported yet
```

Comment:
Should we at least log if we cannot return an error code?

Reply:
Changed to return an error in this case. Addressed in [37ff981](https://github.com/project-chip/connectedhomeip/pull/23840/commits/37ff98112ffb2b2d0dd795ad089f2311d6a417ab).